sample-app/src/components/InputDropdown.tsx

```diff
@@ -104,32 +107,36 @@ export default function InputDropdown({
   function onLeaveSectionFocus(focusNext: boolean) {
     if (focusedSectionIndex === undefined && focusNext) {
       dispatch({ type: 'FocusSection', newIndex: 0 });
-    } else if (focusedSectionIndex !== undefined && focusedSectionIndex < numSections - 1) {
-      let newSectionIndex: number | undefined = focusNext ? focusedSectionIndex + 1 : focusedSectionIndex - 1;
+    } else if (focusedSectionIndex !== undefined) {
+      let newSectionIndex: number | undefined = focusNext
+        ? focusedSectionIndex + 1
+        : focusedSectionIndex - 1;
       if (newSectionIndex < 0) {
         newSectionIndex = undefined;
         onInputChange(latestUserInput);
-      }
-      dispatch({ type: 'FocusSection', newIndex: newSectionIndex });
-    } else if (focusedSectionIndex === numSections - 1 && !focusNext) {
-      let newSectionIndex: number | undefined = focusedSectionIndex - 1;
-      if (newSectionIndex < 0) {
-        newSectionIndex = undefined;
+      } else if (newSectionIndex > numSections - 1) {
+        newSectionIndex = numSections - 1;
       }
       dispatch({ type: 'FocusSection', newIndex: newSectionIndex });
     }
   }
 
   function handleDocumentClick(evt: MouseEvent) {
     const target = evt.target as HTMLElement;
-    if (!(target.isSameNode(inputRef.current) || (dropdownRef.current !== null && dropdownRef.current.contains(target)))) {
+    if (!(target.isSameNode(inputRef.current) || (dropdownRef.current?.contains(target)))) {  
```

Comment:
nit: extra spacing after the curly brackets

Reply:
updated!